### ISSUE TITLE
SSTDPN constrains proto_sep along the wrong axis

### ISSUE THREAD

**Issue by bruAristimunha** [braindecode/braindecode#1059]
## Summary

`SSTDPN.forward` appears to constrain the wrong axis of `proto_sep`.

Braindecode currently renormalizes `proto_sep` with:

```python
self.proto_sep.data = torch.renorm(
    self.proto_sep.data, p=2, dim=1, maxnorm=self.proto_sep_maxnorm
)
```

Given `proto_sep.shape == (n_outputs, feat_dim)`, this constrains feature columns across classes. The paper/source implementation constrains each inter-class separation prototype vector, i.e. each class row.

## Where to patch

`braindecode/models/sstdpn.py`, in `SSTDPN.forward`:

```diff
- self.proto_sep.data = torch.renorm(
-     self.proto_sep.data, p=2, dim=1, maxnorm=self.proto_sep_maxnorm
- )
+ self.proto_sep.data = torch.renorm(
+     self.proto_sep.data, p=2, dim=0, maxnorm=self.proto_sep_maxnorm
+ )
```

## Evidence

In the official SST-DPN source, the inter-class separation prototypes are row vectors and are constrained with `dim=0`:

```python
self.isp.data = torch.renorm(self.isp.data, p=2, dim=0, maxnorm=1)
```

This matches the paper description: each ISP/prototype vector is constrained to `||s_i|| <= S`, with `S=1`.

## Suggested regression test

Instantiate `SSTDPN(n_outputs=4, n_chans=22, n_times=1000)`, set `proto_sep.data.fill_(10)`, run a forward pass, then assert:

```python
torch.all(model.proto_sep.detach().norm(p=2, dim=1) <= model.proto_sep_maxnorm + 1e-6)
```

With the current `dim=1` implementation, row norms are not the constrained quantity.
